time-series/time-series-api/src/main/java/com/powsybl/timeseries/RegularTimeSeriesIndex.java

```diff
@@ -163,7 +163,18 @@ public Duration getTimeStep() {
     }
 
     private static long computePointCount(Instant startTime, Instant endTime, Duration spacing) {
-        return Math.round(((double) (Duration.between(startTime, endTime).toNanos())) / spacing.toNanos()) + 1;
+        // Checks to avoid invalid duration and instants
+        if (startTime == null || endTime == null || spacing == null) {
+            throw new IllegalArgumentException("startTime, endTime, and spacing cannot be null.");
+        }
+
+        Duration duration = Duration.between(startTime, endTime);
+        Long maxDays = 365L * 200;
+        if (duration > maxDays || spacing > maxDays) {
```

Comment:
Not sure you can compare Duration to long
```suggestion
        // Long.MAX_VALUE seconds corresponds to 292 years: Long.MAX_VALUE / (60 * 60 * 24 * 365) = 292.47
        Duration maxDays = Duration.ofDays(365L * 200);
        if (duration.compareTo(maxDays) > 0 || spacing.compareTo(maxDays) > 0) {
```